### PR TITLE
Enabling Full scope training and Add Suite feature

### PR DESCRIPTION
**Pull request by WeizhuoZhang-intel**

### CODE REVIEW

File: groovy/inductor_locally_benchmark.groovy [hunk 1]
```diff
@@ -272,6 +272,7 @@ if( 'specify_image_tag' in params && params.specify_image_tag != '' ) {
 }
 echo "specify_image_tag: $specify_image_tag"
 
+echo "suite: ${suite}"
```
Comment (chuanqi129) [inline groovy/inductor_locally_benchmark.groovy:275]: missed `suite` in this groovy?

File: groovy/inductor_aws_benchmark.groovy [hunk 2]
```diff
@@ -523,19 +528,19 @@ node(NODE_LABEL){
                 #!/usr/bin/env bash
                 cd ${WORKSPACE} && mkdir -p refer && cp -r inductor_log refer && rm -rf inductor_log
                 if [ ${_dash_board} == "true" ]; then
-                     cp scripts/modelbench/report.py ${WORKSPACE} && python report.py -r refer -t ${_target} -m ${_THREADS} --gh_token ${_gh_token} --dashboard ${_dashboard_title} --url ${BUILD_URL} --image_tag ${_target}_aws && rm -rf refer
+                    cp scripts/modelbench/report.py ${WORKSPACE} && python report.py -r refer -t ${_target} -m ${_THREADS} --gh_token ${_gh_token} --dashboard ${_dashboard_title} --url ${BUILD_URL} --image_tag ${_target}_aws && rm -rf refer
```
Comment (chuanqi129) [inline groovy/inductor_aws_benchmark.groovy:531]: Seems no change in this part, right? Not sure, if we just run a single suite whether the report generator can handle, please help to double check it

File: groovy/inductor_nightly.groovy [hunk 3]
```diff
@@ -232,23 +232,23 @@ if ('op_repeats' in params) {
...
-        SHAPE = params.SHAPE
+TEST_SHAPE = 'static'
+if ('TEST_SHAPE' in params) {
+    echo "TEST_SHAPE in params"
+    if (params.TEST_SHAPE != '') {
+        TEST_SHAPE = params.TEST_SHAPE
     }
 }
-echo "SHAPE: $SHAPE"
+echo "TEST_SHAPE: $TEST_SHAPE"
```
Comment (chuanqi129) [inline groovy/inductor_nightly.groovy:251]: seems need update jenkins job parameters name at the same time

File: scripts/aws/docker_prepare.sh [hunk 4]
```diff
@@ -6,6 +6,7 @@ sudo apt install docker.io docker-buildx -y && sudo usermod -aG docker ${USER}
 if [[ ! $(groups "${USER}" | grep -q docker) ]]; then
   sudo groupadd docker
   sudo gpasswd -a ${USER} docker
+  sudo chmod 666 /var/run/docker.sock
```
Comment (chuanqi129) [inline scripts/aws/docker_prepare.sh:9]: Why we need this change?

File: scripts/modelbench/entrance.sh [hunk 5]
```diff
@@ -21,7 +21,7 @@ CHANNELS=${15:-first}
 WRAPPER=${16:-default}
 HF_TOKEN=${17:-hf_xx}
 BACKEND=${18:-inductor}
-SUITE=${19:-torchbench}
+SUITE=${19:-all}
```
Comment (chuanqi129) [inline scripts/modelbench/entrance.sh:24]: it should not impact the guilty commit search part, right?

File: scripts/modelbench/inductor_test.sh [hunk 6]
```diff
@@ -15,18 +15,28 @@ LOG_DIR=${5:-inductor_log}
...
 pip uninstall transformers -y && pip install transformers==4.30.2
+# fix issue: AttributeError: module 'importlib.resources' has no attribute 'files'
+pip uninstall networkx -y && pip install networkx
 
 # skip sam & nanogpt_generate for stable results
 sed -i '/skip_str = " ".join(skip_tests)/a\    skip_str += " -x sam -x nanogpt_generate"' benchmarks/dynamo/runner.py
 
+if [[ ${TEST_MODE} == "training_full" ]]
+    # skip hf_GPT2_large, cuz it will OOM after using jemalloc
+    sed -i "/SKIP_TRAIN = {/a\    \"hf_GPT2_large\"," benchmarks/dynamo/torchbench.py
```
Comment (chuanqi129) [inline scripts/modelbench/inductor_test.sh:37]: whether this list can effect when the test launched by runner.py? And is there no more OOM models in other 2 suites?

File: scripts/modelbench/inductor_test.sh [hunk 7]
```diff
@@ -75,16 +91,17 @@ single_thread_test() {
     if [[ $CHANNELS == "first" ]]; then
         # channels first
         echo "Channels first testing...."
-        python benchmarks/dynamo/runner.py --enable_cpu_launcher --cpu_launcher_args "--core_list 0 --ncores_per_instance 1" --dashboard-archive-path=${LOG_DIR}/archive --devices=cpu --dtypes=${DT} --inference --compilers=${BACKEND} --batch_size=1 --threads 1 --extra-args="--timeout 9000 ${Shape_extra} ${Wrapper_extra} ${Flag_extra}" --output-dir=${LOG_DIR}/single_thread_cf_logs_${timestamp} 2>&1 | tee ${LOG_DIR}/single_thread_model_bench_log_${timestamp}.log
+        python benchmarks/dynamo/runner.py --enable_cpu_launcher --cpu_launcher_args "--core_list 0 --ncores_per_instance 1" --dashboard-archive-path=${LOG_DIR}/archive --devices=cpu --dtypes=${DT} --${TEST_MODE} --compilers=${BACKEND} --batch_size=1 --threads 1 --extra-args="--timeout 9000 ${Shape_extra} ${Wrapper_extra} ${Flag_extra}" --output-dir=${LOG_DIR}/single_thread_cf_logs_${timestamp} 2>&1 | tee ${LOG_DIR}/single_thread_model_bench_log_${timestamp}.log
```
Comment (chuanqi129) [inline scripts/modelbench/inductor_test.sh:94]: Also add `suite` in those 2 cmd, right?

File: scripts/modelbench/launch.sh [hunk 8]
```diff
@@ -85,7 +85,9 @@ if [ $TORCH_START_COMMIT == $TORCH_END_COMMIT ]; then
     docker exec -i $USER bash -c "bash version_collect.sh $LOG_DIR $DYNAMO_BENCH"
 
     if [ $TEST_MODE == "inference" ]; then
-        docker exec -i $USER bash -c "bash inductor_test.sh $THREADS $CHANNELS $PRECISION $TEST_SHAPE $LOG_DIR $WRAPPER $HF_TOKEN $BACKEND $EXTRA"
+        docker exec -i $USER bash -c "bash inductor_test.sh $THREADS $CHANNELS $PRECISION $TEST_SHAPE $LOG_DIR $WRAPPER $HF_TOKEN $BACKEND inference $SUITE $EXTRA"
```
Comment (chuanqi129) [inline scripts/modelbench/launch.sh:88]: also need add `suite` in those 2 lines

**Review by chuanqi129 — COMMENTED** [review]
Overall LGTM, still have minor comments please help to double check them

File: groovy/inductor_aws_benchmark.groovy (hunk 2 above)
Comment (WeizhuoZhang-intel) [reply groovy/inductor_aws_benchmark.groovy:531]: The space used to be 5 spaces, and I fixed it to 4 standard spaces.

File: groovy/inductor_nightly.groovy (hunk 3 above)
Comment (WeizhuoZhang-intel) [reply groovy/inductor_nightly.groovy:251]: I dis change the jenkins job parameters at the same time

File: scripts/aws/docker_prepare.sh (hunk 4 above)
Comment (WeizhuoZhang-intel) [reply scripts/aws/docker_prepare.sh:9]: Cuz if it's a  new docker environment, and our account is a non-root account, we need this permission for a non-root account and the account is in docker group.

File: scripts/modelbench/launch.sh (hunk 8 above)
Comment (WeizhuoZhang-intel) [reply scripts/modelbench/launch.sh:88]: already added at the end of this line.`inference $SUITE $EXTRA"`

File: scripts/modelbench/entrance.sh (hunk 5 above)
Comment (WeizhuoZhang-intel) [reply scripts/modelbench/entrance.sh:24]: Yes, cuz guilty commit job set torchbench as it's default value in Jenkins job parameters already.

File: scripts/modelbench/inductor_test.sh (hunk 6 above)
Comment (WeizhuoZhang-intel) [reply scripts/modelbench/inductor_test.sh:37]: Yes, this model is the only 1 model which will OOM. And I verified this several times before. Cuz runner will read this list `SKIP_TRAIN` before executing training.

File: scripts/modelbench/inductor_test.sh (hunk 7 above)
Comment (WeizhuoZhang-intel) [reply scripts/modelbench/inductor_test.sh]: Done

File: groovy/inductor_locally_benchmark.groovy (hunk 1 above)
Comment (WeizhuoZhang-intel) [reply groovy/inductor_locally_benchmark.groovy:279]: Done

**Comment by chuanqi129** [comment]
Another comment for `suite` parameter, does the report generate part support this scenario?

**Comment by WeizhuoZhang-intel** [comment]
> Another comment for `suite` parameter, does the report generate part support this scenario?

I will test this for training full scope and see how report generated.

File: scripts/aws/docker_prepare.sh (hunk 4 above)
Comment (WeizhuoZhang-intel) [reply scripts/aws/docker_prepare.sh:9]: Currently, nope. This was modified at very early stage when I enabled this path. I could revert this line. This issue might has been fixed by other commit or code.

**Review by chuanqi129 — APPROVED** [review]
@WeizhuoZhang-intel Let's merge this change firstly if there is no issue. For report generate with specified suite part, let's verify it in parallel, if there is issue please fix it in another PR, how do you think?

**Pull request merged**